src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,505 @@
+                    value,
+                    index,
+                    d3
+                        .select(`g[aria-describedby="${value.key}"`)
+                        .selectAll(`.${styles.point}`)[0][index]
+                )
+            )
+            .append("circle")
+            .attr("aria-describedby", value.key)
+            .attr("r", (d) => {
+                const weightAreDefined = utils.isDefined(dataTarget.weight)
+                    ? utils.isDefined(dataTarget.weight.min) &&
+                      utils.isDefined(dataTarget.weight.max)
+                    : false;
+                if (
+                    weightAreDefined &&
+                    d.z !== undefined &&
+                    !utils.isDefined(dataTarget.weight.maxRadius)
+                ) {
+                    return bubbleScale(d.z);
+                }
+                if (utils.isUndefined(dataTarget.weight)) {
+                    return constants.DEFAULT_BUBBLE_RADIUS_MAX;
+                }
+                return dataTarget.weight.maxRadius;
+            })
+            .style("fill", (d) => {
+                const hueIsDefined = utils.isDefined(dataTarget.hue);
+                const weightAreDefined = utils.isDefined(dataTarget.weight)
+                    ? utils.isDefined(dataTarget.weight.min) &&
+                      utils.isDefined(dataTarget.weight.max)
+                    : false;
+                const useYAxisData = !utils.isDefined(d.z);
+
+                if (
+                    hueIsDefined &&
+                    weightAreDefined &&
+                    useYAxisData === false
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData
+                    );
+                    return colorHue.get(bubbleScale(d.z));
+                }
+                if (
+                    hueIsDefined &&
+                    weightAreDefined === false &&
+                    useYAxisData
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData
+                    );
+                    return colorHue.get(d.y);
+                }
+                return dataTarget.color;
+            })
+            .style("opacity", "0.8")
+            .attr("stroke", "#d8d8c2");
+    };
+
+    const renderSelectionPath = (path, value, index) => {
+        path.append("g")
+            .classed(styles.dataPointSelection, true)
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-disabled", false)
+            .attr("aria-hidden", true)
+            .attr("aria-describedby", value.key)
+            .append("circle")
```

Comment:
Is there a reason we arent using SVG circle for this?

Reply:
I started with using SVG but it was hard to scale them and for bubble the standard is to use circle so I followed the standard then.